Build fails with `xtensa-esp32s3-espidf` target

### Bug Description

I noticed that when compiling slint targeting `xtensa-esp32s3-espidf`, it failed with the following error.
```
error[E0308]: mismatched types
   --> /home/nazo/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/i-slint-core-1.15.1/graphics/brush.rs:398:91
    |
398 |                 let rotated_position = (stop.position + normalized_from_angle).rem_euclid(&1.0);
    |                                                                                ---------- ^^^^ expected `f32`, found `&{float}`
    |                                                                                |
    |                                                                                arguments to this method are incorrect
    |
note: method defined here
   --> /home/nazo/.rustup/toolchains/esp/lib/rustlib/src/rust/library/std/src/num/f32.rs:288:12
    |
288 |     pub fn rem_euclid(self, rhs: f32) -> f32 {
    |            ^^^^^^^^^^
help: consider removing the borrow
    |
398 -                 let rotated_position = (stop.position + normalized_from_angle).rem_euclid(&1.0);
398 +                 let rotated_position = (stop.position + normalized_from_angle).rem_euclid(1.0);
    |

For more information about this error, try `rustc --explain E0308`.
error: could not compile `i-slint-core` (lib) due to 1 previous error
```
This appears to occur when disabling the std feature in i-slint-core for targets where std is available.

The availability of std introduces rem_euclid to f32, which takes precedence over the trait implementation in num_traits, resulting in a type mismatch.

Probably related to #10855

### Reproducible Code (if applicable)

```slint

```

### Environment Details

- Slint Version: 1.15
- Platform/OS: Linux (Host), esp32s3 IDF (Target)
- Programming Language: Rust
- Backend/Renderer: Software

### Product Impact

_No response_

--- a/internal/core/graphics/brush.rs
+++ b/internal/core/graphics/brush.rs
@@ -11,8 +11,6 @@ use crate::properties::InterpolatedPropertyValue;
 use euclid::default::{Point2D, Size2D};
 
 #[cfg(not(feature = "std"))]
-use num_traits::Euclid;
-#[cfg(not(feature = "std"))]
 use num_traits::float::Float;
 
 /// A brush is a data structure that is used to describe how
@@ -392,10 +390,10 @@ impl ConicGradientBrush {
         stops = stops
             .iter()
             .map(|stop| {
-                #[cfg(feature = "std")]
-                let rotated_position = (stop.position + normalized_from_angle).rem_euclid(1.0);
-                #[cfg(not(feature = "std"))]
-                let rotated_position = (stop.position + normalized_from_angle).rem_euclid(&1.0);
+                // f32::rem_euclid is not always available, and when it is it has a
+                // different signature than num_traits::Euclid::rem_euclid (issue #11333).
+                let rotated_position =
+                    num_traits::Euclid::rem_euclid(&(stop.position + normalized_from_angle), &1.0);
                 GradientStop { position: rotated_position, color: stop.color }
             })
             .collect();

